### DISCUSSION TITLE
Is possible to use an external Supabase auth project with another Supabase project?

### DISCUSSION THREAD

**Discussion by filippobrigati** [supabase/supabase#13427] [Questions]
Let's assume that we have two Supabase projects, one stores only user data for the authentication and another stores some data related to the user BUT that doesn't have the authentication but the authentication should be done throw the Supabase project with auth data.

How should something like that should be done? Should I only change GOTRUE_DB_DATABASE_URL with parameters from the Supabase auth project?

I set the same JWT_SECRET for both project so now i have the same ANON_KEY, but when i try to create a supabase client I get this error: `{"message":"Invalid API key","hint":"Double check your Supabase "anon" or "service_role" API key."}`

someone can help me?

**Comment by GaryAustin1** [comment]
The anon key at some point added the instance reference to the jwt.   Are you sure the anon key is the same for both?  If that is being checked then you would need to use the correct anon key on each instance.

Your user jwt's should work with RLS auth.uid() and auth.jwt().
Storage will NOT likely work if you have individual user files except on the instance with the auth.users table as there is an fk link.

**Reply by filippobrigati** [reply]
Yes I fixed it changing the anon key, now it get me 404 user not found.

It is possible to use Supabase database without users? becouse in my case the users will be in my supabase users database and not in the data one.

**Reply by GaryAustin1** [reply]
It is possible but you don't use auth.  You just set the user jwt in the authorization header for your database calls.

**Reply by filippobrigati** [reply]
but how should i create supabase client from my supabase data db in my frontend? if I use the access_token and the refresh_token of the authentication db it always returns me 404.

Sorry for the dumb question but it's the first time that i'm using supabase.

Thanks for your answers

**Reply by GaryAustin1** [reply]
You can't use refresh token on the database instance.  All you can do is set up a client with the header option to set authorization header to the user jwt (access token).   Then that client will obey RLS per the user jwt.

You would have to deal with expiring and refreshing back on the auth instance.

You can run more than one client at a time by setting the storageKey to a different value in your createClient.  You also need to set persistSession, autoRefreshToken and detectSessionInURL to false in your "database" client.
![image](https://user-images.githubusercontent.com/54564956/229858221-9c48491e-b51b-4108-bf85-2530662ce92c.png)

**Reply by filippobrigati** [reply]
> You can't use refresh token on the database instance.  All you can do is set up a client with the header option to set authorization header to the user jwt (access token).   Then that client will obey RLS per the user jwt.
> 
> 
> 
> You would have to deal with expiring and refreshing back on the auth instance.
> 
> 
> 
> You can run more than one client at a time by setting the storageKey to a different value in your createClient.  You also need to set persistSession, autoRefreshToken and detectSessionInURL to false in your "database" client.
> 
> ![image](https://user-images.githubusercontent.com/54564956/229858221-9c48491e-b51b-4108-bf85-2530662ce92c.png)
> 
> 

Thank you so much
